### PR TITLE
Fixes #25166 - Add Applicable errata report

### PR DESCRIPTION
**Pull request by mbacovsky**

### CODE REVIEW

**Comment by mbacovsky** [comment]
PR in Katello https://github.com/Katello/katello/pull/7747 is required for this report.
I'd like to ask reviewer to check if the queries could be further optimised and if the data collected match what was requested.

File: report_templates/applicable_errata.erb [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  required: true
+  input_type: user
+  description: Format of the generated report (csv or yaml)
+  options: "csv\r\nyaml"
+  advanced: false
+model: ReportTemplate
+require:
+- plugin: katello
+  version: 3.9.0
+%>
```
Comment (ares) [inline report_templates/applicable_errata.erb]: missing `-`

File: report_templates/applicable_errata.erb [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+  description: Limit the report only on hosts found by this search query. Keep empty
+    for report on all available hosts.
+  advanced: false
+- name: Errata filter
+  required: false
+  input_type: user
+  description: Limit the report only on errata found by this search query. Keep empty
+    for report on all available errata.
+  advanced: false
+- name: Output format
```
Comment (ares) [inline report_templates/applicable_errata.erb]: as suggested elsewhere, format should not be in input I think

File: report_templates/applicable_errata.erb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  input_type: user
+  description: Format of the generated report (csv or yaml)
+  options: "csv\r\nyaml"
+  advanced: false
+model: ReportTemplate
+require:
+- plugin: katello
+  version: 3.9.0
+%>
+<%- load_hosts(search: input('Hosts filter'), includes: [:operatingsystem,:applicable_errata,:lifecycle_environment]).each_record do |host| -%>
```
Comment (ares) [inline report_templates/applicable_errata.erb]: please add spaces after commas in array

**Review by ares — CHANGES_REQUESTED** [review]
few smaller things, meanwhile I'm building an environment where I'll be able to test

**Comment by mbacovsky** [comment]
@ares, updated

**Review by ares — APPROVED** [review]

**Comment by ares** [comment]
pending Katello PR merge

**Comment by ares** [comment]
Thanks @mbacovsky, merging!

**Pull request merged**